enhancements/cloud-integration/infrastructure-external-platform-type.md

```diff
@@ -0,0 +1,588 @@
+`CredentialsRequest`s allow OpenShift components to request credentials for a particular cloud provider.
+
+On unsupported platforms, the operator goes into no-op mode, which technically is mostly an equivalent
+of the ["Manual"](https://docs.openshift.com/container-platform/4.11/authentication/managing_cloud_provider_credentials/cco-mode-manual.html)
+mode.
+
+Cloud Credential operator [uses](https://github.com/openshift/cloud-credential-operator/blob/master/pkg/operator/controller.go#L74) an "actuator pattern"
+and, in theory, in the future, it might be extended in a way to react to the new "External" platform type and allow users to supplement their own
+platform-specific credentials management logic.
+
+During initial enablement phases of "External" platform type, no specific actions will be needed there, since
+CCO [would](https://github.com/openshift/cloud-credential-operator/blob/58c41771a87e613415a1fa16d299601a1c2f48c2/pkg/operator/controller.go#L134) go
+into no-op mode if it encounters an unrecognized platform.
+
+#### Cluster Image Registry Operator
+
+For image registry a [storage backend config decision](https://github.com/openshift/cluster-image-registry-operator/blob/99474318db709a6d17d06468b90cdf0dc0fd2b87/pkg/storage/storage.go#L157) is platform specific.
+With the "None" platform type CIRO goes into no-op state, which means that no registry will be deploy in such case.
+The image registry configures with EmptyDir storage for unknown platform type at the moment.
+
+Image Registry storage options will be configured to use PVC-backed or external storage systems
+(such as Ceph or S3 compatible object storage) as a day two operation.
+
+For now, it seems that no particular action for the "External" platform type is needed within the Image Registry Operator,
+since we're providing enough possibilities to customize Image Registry storage backend.
+
+#### Cluster Ingress Operator
+
+Within the ingress operator, a PlatformType affects two things:
+1. Choosing EndpointPublishingStrategy, which is `HostNetworkStrategyType` for "None" and unknown PlatformType
+2. Creating DNS provider on some platforms. This logic does not engage for "None" or unknown PlatformType. The DNS provider is used to create a wildcard DNS record for ingress when using `LoadBalancerServiceStrategyType`. It is not used for `HostNetworkStrategyType`.
+
+With regard to the [EndpointPublishingStrategy](https://docs.openshift.com/container-platform/4.11/networking/ingress-operator.html#nw-ingress-controller-configuration-parameters_configuring-ingress),
+the cluster admin can configure Cluster Ingress Operator to use `LoadBalancerServiceStrategyType` as a day two operation.
+The operator itself creates `Service` objects with correct provider-specific annotations,
+the actual handling of such objects happens in a provider-specific [cloud controller manager](https://kubernetes.io/docs/concepts/architecture/cloud-controller/).
```

Comment:
Reading through this EP today, I realize that it provides fairly specific descriptions of whether/what changes will required for other components, but in contrast, the EP leaves it open-ended for the cluster-ingress-operator component.  However, I figure that that's intentional—in future phases, we can hash out how DNS and LB configuration will be managed and what role cluster-ingress-operator will play for clusters with the "External" platform-type.  If that's true, then the context provided here (without specific solutions) is sufficient for now.

Reply:
i think you discerned the deeper meaning. the cluster-ingress-operator provides enough options that we aren't completely sure if there will be prescriptive advice we can give about changes that might come in phase 3. we'd like to re-evaluate, with the network team, when and if users start pressing for ingress options that might not be currently available.